opm/autodiff/SimulatorBase_impl.hpp

```diff
@@ -258,14 +258,20 @@ namespace Opm
             if (i != wmap.end()) {
                 WellConstPtr wp = i->second;
 
+                // Note that RESV is always a valid control
+                // for history matching producers.
                 match = (wp->isProducer(step) &&
                          wp->getProductionProperties(step)
                          .hasProductionControl(WellProducer::RESV))
                     ||  (wp->isInjector(step) &&
                          wp->getInjectionProperties(step)
-                         .hasInjectionControl(WellInjector::RESV));
+                         .hasInjectionControl(WellInjector::RESV))
+                    ||   (wp->isInjector(step) &&
+                          !wp->getInjectionProperties(step).predictionMode);
```

Comment:
I do not understand this: does being a history-matching injector automatically mean that we have a RESV control?